src/detect.h

```diff
@@ -462,7 +485,8 @@ typedef struct DetectBufferType_ {
     bool supports_transforms;
     bool multi_instance; /**< buffer supports multiple buffer instances per tx */
     void (*SetupCallback)(const struct DetectEngineCtx_ *, struct Signature_ *);
-    bool (*ValidateCallback)(const struct Signature_ *, const char **sigerror);
+    bool (*ValidateCallback)(
```

Comment:
we need to know `DetectContentData` otherwise compiler is not happy with
```
declaration of 'struct DetectContentData_' will not be visible outside of this function [-Wvisibility]
```